src/lightning_habana/pytorch/datamodule/datamodule.py

```diff
@@ -0,0 +1,210 @@
+elif module_available("pytorch_lightning"):
+    import pytorch_lightning as pl
+    from pytorch_lightning.utilities.imports import _TORCHVISION_AVAILABLE
+
+import torch
+
+from lightning_habana.utils.imports import _HPU_AVAILABLE
+
+if _TORCHVISION_AVAILABLE:
+    import torchvision.datasets
+    from torchvision import transforms as transform_lib
+
+if _HPU_AVAILABLE:
+    try:
+        import habana_dataloader
+    except ImportError:
+        raise ModuleNotFoundError("habana_dataloader package is not installed.")
+
+    from lightning_habana.pytorch.datamodule.dataloaders.resnet_media_pipe import MediaApiDataLoader
+
+import lightning_habana.pytorch.datamodule.utils
+
+_DATASETS_PATH = "/tmp/data"
+
+
+def load_data(traindir, valdir, train_transforms, val_transforms):  # type: ignore[no-untyped-def]
+    """Helper to initialize dataset and transforms."""
+    # check supported transforms
+    if train_transforms is not None:
+        for t in train_transforms:
+            if isinstance(
+                t,
+                (
+                    transform_lib.RandomResizedCrop,
+                    transform_lib.RandomHorizontalFlip,
+                    transform_lib.ToTensor,
+                    transform_lib.Normalize,
+                ),
+            ):
+                print("Train Transforms supported")
+            else:
+                raise ValueError("Unsupported train transform: " + str(type(t)))
+
+        train_transforms = transform_lib.Compose(train_transforms)
+
+    if val_transforms is not None:
+        for t in val_transforms:
+            if isinstance(
+                t, (transform_lib.Resize, transform_lib.CenterCrop, transform_lib.ToTensor, transform_lib.Normalize)
+            ):
+                print("Val Transforms supported")
+            else:
+                raise ValueError("Unsupported val transform: " + str(type(t)))
+
+        val_transforms = transform_lib.Compose(val_transforms)
+
+    if "imagenet" not in traindir.lower() and "ilsvrc2012" not in traindir.lower():
+        raise ValueError("Habana dataloader only supports Imagenet dataset")
+
+    # Data loading code
+    dataset_train = torchvision.datasets.ImageFolder(traindir, train_transforms)
+
+    dataset_val = torchvision.datasets.ImageFolder(valdir, val_transforms)
+
+    return dataset_train, dataset_val
+
+
+class HPUDataModule(pl.LightningDataModule):
+    """Datamodule helper class to load the right media pipe."""
+
+    name = "hpu-dataset"
+
+    def __init__(
+        self,
+        train_dir: str = _DATASETS_PATH,
+        val_dir: str = _DATASETS_PATH,
+        num_workers: int = 8,
+        normalize: bool = False,
+        seed: int = 42,
+        batch_size: int = 32,
+        train_transforms: Any = None,
+        val_transforms: Any = None,
+        pin_memory: bool = True,
+        shuffle: bool = False,
+        drop_last: bool = True,
+        dl_type: str = "MP",
```

Comment:
"MP" is just to show case the mechanics for user to enable native dataloader. Default will be selected based on hardware below

Reply:
I think it is backwards. If I want to use this datamodule, by default it will have dl_type = "MP", so if I create a datamodule like this:

```py
datamodule = HPUDataModule()
```
it will select the torch data loader. If I want to use the the HabanaDataLoader, I have to pass in a random string, for example:

```py
datamodule = HPUDataModule(dl_type="coconut")
```
Now it will use the habana loader. I think this is very strange.

Reply:
The logic in lines 129-137 should probably be changed.

Reply:
Will remove the confusing parts